Source/DafnyDriver/Legacy/SynchronousCliCompilation.cs

```diff
@@ -162,20 +162,19 @@ public class SynchronousCliCompilation : IDisposable {
         } else if (options.AllowSourceFolders && Directory.Exists(file)) {
           options.SourceFolders.Add(file);
         } else if (!isDafnyFile) {
+          string errorOnNotRecognized;
           if (options.UsingNewCli && string.IsNullOrEmpty(extension) && file.Length > 0) {
-            options.Printer.ErrorWriteLine(options.OutputWriter,
-              "*** Error: Command-line argument '{0}' is neither a recognized option nor a filename with a supported extension ({1}).",
-              nameToShow,
-              string.Join(", ", Enumerable.Repeat(".dfy", 1).Concat(supportedExtensions)));
+            errorOnNotRecognized =
+              $"*** Error: Command-line argument '{nameToShow}' is neither a recognized option nor a filename with a supported extension ({string.Join(", ", Enumerable.Repeat(".dfy", 1).Concat(supportedExtensions))}).";
           } else if (string.IsNullOrEmpty(extension) && file.Length > 0 && (file[0] == '/' || file[0] == '-')) {
-            options.Printer.ErrorWriteLine(options.OutputWriter,
-              "*** Error: Command-line argument '{0}' is neither a recognized option nor a filename with a supported extension ({1}).",
-              nameToShow, string.Join(", ", Enumerable.Repeat(".dfy", 1).Concat(supportedExtensions)));
+            errorOnNotRecognized =
+              $"*** Error: Command-line argument '{nameToShow}' is neither a recognized option nor a filename with a supported extension ({string.Join(", ", Enumerable.Repeat(".dfy", 1).Concat(supportedExtensions))}).";
           } else {
-            options.Printer.ErrorWriteLine(options.OutputWriter,
-              "*** Error: '{0}': Filename extension '{1}' is not supported. Input files must be Dafny programs (.dfy) or supported auxiliary files ({2})",
-              nameToShow, extension, string.Join(", ", supportedExtensions));
+            errorOnNotRecognized =
+              $"*** Error: '{nameToShow}': Filename extension '{extension}' is not supported. Input files must be Dafny programs (.dfy) or supported auxiliary files ({string.Join(", ", supportedExtensions)})";
           }
+
+          options.Printer.ErrorWriteLine(options.OutputWriter, errorOnNotRecognized);
```

Comment:
I'm surprised you didn't make this a proper ErrorReporter.Error call instead, or is that not possible at this point in the flow?

Reply:
Just kept the old code. I'm not too worried about this file since I plan on replacing it. Also, I don't think there's an `ErrorReporter` available here.